src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,156 @@
+                }
+            }
+
+            Address = address;
+            _transportType = transportType;
+            _portType = portType;
+        }
+
+        public static bool TryParse(string config, out IpcEndpointConfig result)
+        {
+            try
+            {
+                result = Parse(config);
+            }
+            catch(Exception)
+            {
+                result = null;
+            }
+            return result != null;
+        }
+
+        public static IpcEndpointConfig Parse(string config)
+        {
+            if (string.IsNullOrEmpty(config))
+                throw new FormatException("Missing IPC endpoint config.");
+
+            string address = "";
+            PortType portType = PortType.Connect;
+            TransportType transportType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? TransportType.NamedPipe : TransportType.UnixDomainSocket;
+
+            if (!string.IsNullOrEmpty(config))
+            {
+                var parts = config.Split(',');
+                if (parts.Length > 2)
+                    throw new FormatException($"Unknow IPC endpoint config format, {config}.");
+
+                if (string.IsNullOrEmpty(parts[0]))
+                    throw new FormatException($"Missing IPC endpoint config address, {config}.");
+
+                portType = PortType.Listen;
+                address = parts[0];
+
+                if (parts.Length == 2)
+                {
+                    if (string.Equals(parts[1], "connect", StringComparison.OrdinalIgnoreCase))
+                    {
+                        portType = PortType.Connect;
+                    }
+                    else if (string.Equals(parts[1], "listen", StringComparison.OrdinalIgnoreCase))
+                    {
+                        portType = PortType.Listen;
+                    }
+                    else
+                    {
+                        throw new FormatException($"Unknow IPC endpoint config keyword, {parts[1]} in {config}.");
+                    }
+                }
+            }
+
+            if (Uri.TryCreate(address, UriKind.Absolute, out Uri parsedAddress))
+            {
+                if (string.Equals(parsedAddress.Scheme, NamedPipeSchema, StringComparison.OrdinalIgnoreCase))
+                {
+                    transportType = TransportType.NamedPipe;
+                    address = parsedAddress.AbsolutePath;
+                }
+                else if (string.Equals(parsedAddress.Scheme, UnixDomainSocketSchema, StringComparison.OrdinalIgnoreCase))
+                {
+                    transportType = TransportType.UnixDomainSocket;
+                    address = parsedAddress.AbsolutePath;
+                }
+                else if (!string.IsNullOrEmpty(parsedAddress.Scheme))
+                {
+                    throw new FormatException($"{parsedAddress.Scheme} not supported.");
+                }
```

Comment:
```suggestion
                (transportType, address) = parsedAddress.Scheme.ToLower() switch
                {
                	NamedPipeSchema => (TransportType.NamedPipe, parsedAddress.AbsolutePath),
                	UnixDomainSocketSchema => (TransportType.UnixDomainSocket, parsedAddress.AbsolutePath),
                	_ => throw new FormatException($"{parsedAddress.Scheme} not supported.");
                };
```

Okay, I'll stop with the switch expression suggestions 😄

Reply:
I will keep the if/else if/else for now.